src/components/Input/Input.js

```diff
@@ -182,23 +209,11 @@ const Input = ({
 
   const suffixClassName = cx(suffixStyles({ theme }));
 
-  /* eslint-disable react/prop-types */
-  const newSuffix = (() => {
-    if (invalid) {
-      return ({ className }) => <ErrorIcon {...{ className }} />;
-    }
-    if (hasWarning) {
-      return ({ className }) => <WarningIcon {...{ className }} />;
-    }
-    if (showValid) {
-      return ({ className }) => <ValidIcon {...{ className }} />;
-    }
-    return suffix;
-  })();
-  /* eslint-enable react/prop-types */
+  const suffix =
+    customSuffix || determineSuffix({ invalid, hasWarning, showValid });
```

Comment:
Just realized this now, but I think the way `determineSuffix` works at the moment, you'd get a runtime error here, if the function just returns `null`.

Reply:
Nope. 😊  The component first checks if the suffix exists at all before calling it.